tests/unit_tests/template.py

```diff
@@ -0,0 +1,834 @@
+import json
+import re
+import subprocess
+import os
+import sqlite3 as lite
+import copy
+
+from nose.tools import assert_in, assert_true, assert_equals
+
+direc = os.listdir('./')
+for item in direc:
+    if item.endswith('.sqlite') or item.endswith('.json'):
+        os.remove(os.path.join('./', item))
+
+env = dict(os.environ)
+env['PYTHONPATH'] = "."
+
+def get_cursor(file_name):
+    """ Connects and returns a cursor to an sqlite output file
+
+    Parameters
+    ----------
+    file_name: str
+        name of the sqlite file
+
+    Returns
+    -------
+    sqlite cursor3
+    """
+    con = lite.connect(file_name)
+    con.row_factory = lite.Row
+    return con.cursor()
+
+def cleanup():
+    """ Removes the generated input file and output sqlite database file."""
+    if os.path.exists(output_file):
+        os.remove(output_file)
+    if os.path.exists(input_file):
+        os.remove(input_file)
+
+# the template is a dictionary of all simulation parameters
+# except the region-institution definition
+template = {
+ "simulation": {
+  "archetypes": {
+   "spec": [
+    {"lib": "agents", "name": "NullRegion"}, 
+    {"lib": "cycamore", "name": "Source"}, 
+    {"lib": "cycamore", "name": "Reactor"},
+    {"lib": "cycamore", "name": "Sink"},
+    {"lib": "d3ploy.no_inst", "name": "NOInst"}
+   ]
+  }, 
+  "control": {"duration": "15", "startmonth": "1", "startyear": "2000"}, 
+  "recipe": [
+   {
+    "basis": "mass", 
+    "name": "fresh_uox", 
+    "nuclide": [{"comp": "0.711", "id": "U235"}, {"comp": "99.289", "id": "U238"}]
+   }, 
+   {
+    "basis": "mass", 
+    "name": "spent_uox", 
+    "nuclide": [{"comp": "50", "id": "Kr85"}, {"comp": "50", "id": "Cs137"}]
+   }
+  ], 
+  "facility": [{
+   "config": {"Source": {"outcommod": "fuel",
+                         "outrecipe": "fresh_uox",
+                         "throughput": "1",
+                         "source_record_supply": "fuel"}}, 
+   "name": "source"
+  },
+  {
+   "config": {"Sink": {"in_commods": {"val":"spent_uox"},
+                       "max_inv_size": 1,
+                       "sink_record_demand": "fuel_cap"}}, 
+   "name": "sink"
+  },
+  {
+   "config": {
+        "Reactor":{
+            "assem_size":"1",
+            "cycle_time": "1", 
+            "fuel_incommods": {"val": "fuel"}, 
+            "fuel_inrecipes": {"val": "fresh_uox"}, 
+            "fuel_outcommods": {"val": "spent_uox"}, 
+            "fuel_outrecipes": {"val": "spent_uox"}, 
+            "n_assem_batch": "1", 
+            "n_assem_core": "1", 
+            "power_cap": "1", 
+            "refuel_time": "0",
+            "reactor_fuel_demand": "fuel_reactor"
+        }
+    },
+    "name": "reactor"
+   }]
+  }
+}
+
+""" testA refers to if NOInst meets the increased demand by deploying more facilities.
+    testB refers to if NOInst meets the decreased demand by decommissioning existing facilities.
+"""
+
+""" TestA Examples"""
```

Comment:
It is not clear what these strings are doing here. Are these supposed to be comments? If so, please make them comments